src/mpid/ch4/shm/posix/eager/iqueue/iqueue_init.c

```diff
@@ -167,8 +163,12 @@ int MPIDI_POSIX_iqueue_set_vcis(MPIR_Comm * comm)
         }
     }
 
-    mpi_errno = MPIDU_Init_shm_barrier();
-    MPIR_ERR_CHECK(mpi_errno);
+    MPIDI_POSIX_eager_iqueue_global.max_vcis = max_vcis;
+
+    if (comm->node_comm) {
+        mpi_errno = MPIR_Barrier_impl(comm->node_comm, MPIR_ERR_NONE);
+        MPIR_ERR_CHECK(mpi_errno);
+    }
```

Comment:
Is it necessary to change from `MPIDU_Init_shm_barrier` here? For the single process case maybe?